### PR TITLE
Laboratory work #2, Maxim Krivenko - 23FPL2

### PR DESCRIPTION
**Pull request by makskrivenko**

### CODE REVIEW

File: lab_2_retrieval_w_bm25/settings.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-    "target_score": 0
+    "target_score": 4
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/settings.json:2]: Where is start.py?  And what about 6, 8, 10?

Comment (makskrivenko) [reply lab_2_retrieval_w_bm25/settings.json:2]: work in progress

File: lab_2_retrieval_w_bm25/main.py [hunk 2]
```diff
@@ -18,6 +18,16 @@ def tokenize(text: str) -> list[str] | None:
...
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(text, str):
+        return None
+
+    for symbol in text:
+        if not symbol.isalpha():
+            text = text.replace(symbol, ' ')
+
+    tokens = text.lower().split()
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:28]: do not create the variable

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_retrieval_w_bm25/main.py (hunk 2 above)
Comment (makskrivenko) [reply lab_2_retrieval_w_bm25/main.py]: fixed

File: lab_2_retrieval_w_bm25/main.py [hunk 3]
```diff
@@ -62,6 +93,20 @@ def calculate_tf(vocab: list[str], document_tokens: list[str]) -> dict[str, floa
...
+    if not ((isinstance(vocab, list) and vocab and all(isinstance(word_1, str) for word_1 in vocab)) and
+       (isinstance(document_tokens, list) and document_tokens and
+       all(isinstance(word_2, str) for word_2 in document_tokens))):
+        return None
+
+    all_words = set(vocab).union(document_tokens)
+    tf_dict = {}
+    for t_word in all_words:
+        if t_word in document_tokens:
+            tf_dict[t_word] = document_tokens.count(t_word) / len(document_tokens)
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:105]: do it always, but set 0 only if not exists

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_retrieval_w_bm25/main.py (hunk 3 above)
Comment (makskrivenko) [reply lab_2_retrieval_w_bm25/main.py:105]: done

**Pull request closed without merging**